hoptimator-operator/src/main/java/com/linkedin/hoptimator/operator/subscription/SubscriptionReconciler.java

```diff
@@ -106,6 +150,102 @@ Pipeline pipeline(V1alpha1Subscription object) throws Exception {
+          namespace, name, ownerReference.getKind(), ownerReference.getName());
+      } else {
+        log.info("Existing downstream resource {}/{} will be owned by {}/{} and {} others.",
+          namespace, name, ownerReference.getKind(), ownerReference.getName(), owners.size());
+        owners.add(ownerReference);
+      }
+      obj.setMetadata(obj.getMetadata().ownerReferences(owners).resourceVersion(resourceVersion));
+      KubernetesApiResponse<DynamicKubernetesObject> response = operator.apiFor(obj).update(obj);
+      if (!response.isSuccess()) {
+        log.error("Error updating downstream resource {}/{}: {}.", namespace, name, response.getStatus().getMessage());
+        return false;
+      }
+    } else {
+      log.info("Creating downstream resource {}/{} as \n{}", namespace, name, yaml);
+      obj.setMetadata(obj.getMetadata().addOwnerReferencesItem(ownerReference));
+      KubernetesApiResponse<DynamicKubernetesObject> response = operator.apiFor(obj).create(obj);
+      if (!response.isSuccess()) {
+        log.error("Error creating downstream resource {}/{}: {}.", namespace, name, response.getStatus().getMessage());
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean checkStatus(String yaml) {
+    DynamicKubernetesObject obj = Dynamics.newFromYaml(yaml);
+    String namespace = obj.getMetadata().getNamespace();
+    String name = obj.getMetadata().getName();
+    String kind = obj.getKind();
+    try {
+      KubernetesApiResponse<DynamicKubernetesObject> existing = operator.apiFor(obj).get(namespace, name);
+      existing.onFailure((code, status) -> log.warn("Failed to fetch {}/{}: {}.", kind, name, status.getMessage()));
+      if (!existing.isSuccess()) {
+        return false;
+      }
+      if (isReady(existing.getObject())) {
+        log.info("{}/{} is ready.", kind, name);
+        return true;
+      } else {
+        log.info("{}/{} is NOT ready.", kind, name);
+        return false;
+      }
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  private static boolean isReady(DynamicKubernetesObject obj) {
+    // We make a best effort to guess the status of the dynamic object. By default, it's ready.
+    if (obj == null || obj.getRaw() == null) {
+      return false;
+    }
+    try {
+      return obj.getRaw().get("status").getAsJsonObject().get("ready").getAsBoolean();
+    } catch (Exception e) {
+      log.debug("Exception looking for .status.ready. Swallowing.", e);
+    }
+    try {
+      return obj.getRaw().get("status").getAsJsonObject().get("state").getAsString()
+        .matches("(?i)READY|RUNNING|FINISHED");
+    } catch (Exception e) {
+      log.debug("Exception looking for .status.state. Swallowing.", e);
+    }
+    try {
+      return obj.getRaw().get("status").getAsJsonObject().get("jobStatus").getAsJsonObject()
+        .get("state").getAsString().matches("(?i)READY|RUNNING|FINISHED");
+    } catch (Exception e) {
```

Comment:
[Minor]: If I understand this correctly, all these different try catch blocks are so that we can have a custom log? Would it be possible to unify them and since you are anyway logging exception object `e` we will have idea on what is causing this, especially where we are catching only Exception, there is no particular runtime exception type we are catching which needs separate catch block.

Reply:
The idea here is to try a bunch of ways to check the status of a generic resource, which is difficult since there is no conventional way to report readiness. The multiple try blocks is so we don't need a bunch of `if (contains(x))` statements.

Reply:
Ok, so each downstream resource has their own format. would it be a good idea to namespace them, like if resource is x -> use this jsonPath? 

Can we make the matcher string `"(?i)READY|RUNNING|FINISHED"` constant?